master/buildbot/db/buildrequests.py

```diff
@@ -111,9 +112,24 @@ def thd(conn):
                 q = q.where(reqs_tbl.c.buildsetid == bsid)
 
             if branch is not None:
-                q = q.where(sstamps_tbl.c.branch == branch)
+                q = q.column(sstamps_tbl.c.branch).where(sstamps_tbl.c.branch == branch)
             if repository is not None:
-                q = q.where(sstamps_tbl.c.repository == repository)
+                q = q.column(sstamps_tbl.c.repository).where(sstamps_tbl.c.repository == repository)
+
+            # ONLY_FULL_GROUP_BY enabled by default in MySQL obliges us to indicate all the group_by columns
```

Comment:
obliges looks franglish :) requires looks better to me. That said, it is not clear to me why that huge group_by is needed.
Isn't distinct more appropriate?
http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.distinct

Reply:
I thought about that yesterday, but unfortunately using dinstinct() had no effect :/